src/rcc.rs

```diff
@@ -252,6 +254,65 @@ impl CFGR {
     }
 }
 
+/// Control/Status Register
+pub struct CSR(());
+
+impl CSR {
+    fn csr(&mut self) -> &rcc::CSR {
+        // NOTE(unsafe) this proxy grants exclusive access to this register
+        unsafe { &(*RCC::ptr()).csr }
+    }
+
+    /// Enable the LSI clock around 40 kHz (30..60 kHz)
+    ///
+    /// Needed for the Independent Watchdog (IWDG) periphal and the
+    /// Auto-Wakeup Unit (AWU)
+    pub fn enable_lsi(&mut self) {
+        // Enable
+        self.csr().modify(|_, w| w.lsion().set_bit());
+
+        // Wait for clock to become stable
+        while self.csr().read().lsirdy().bit_is_set() {}
+    }
+
+    /// Read why a reset occured
+    pub fn reset_reason(&mut self) -> ResetReason {
+        let r = self.csr().read();
+        if r.lpwrrstf().bit_is_set() {
+            ResetReason::LowPower
+        } else if r.wwdgrstf().bit_is_set() {
+            ResetReason::WWDG
+        } else if r.iwdgrstf().bit_is_set() {
+            ResetReason::IWDG
+        } else if r.sftrstf().bit_is_set() {
+            ResetReason::Software
+        } else if r.porrstf().bit_is_set() {
+            ResetReason::PorPdr
+        } else if r.pinrstf().bit_is_set() {
+            ResetReason::NrstPin
+        } else {
+            ResetReason::Unknown
+        }
+    }
+
+    /// Clear the reason for the last reset
+    pub fn clear_reset_reason(&mut self) {
+        self.csr().write(|w| w.rmvf().set_bit());
+    }
+}
+
+/// Reason for controller reset
+#[derive(Clone, Copy, Debug)]
+pub enum ResetReason {
```

Comment:
How about adding documentation on the variants of this enum? I would say some are not immediately obvious.